### PR TITLE
Add sherpa-onnx support

### PR DESCRIPTION
**Pull request by Neil2893**
This pull request adds support for sherpa-onnx (https://github.com/k2-fsa/sherpa-onnx) ASR models. It allows users to easily integrate various sherpa-onnx models, including transducer, Paraformer, NeMo CTC, WeNet CTC, Whisper, TDNN CTC, and SenseVoice models.

Note: Only SenseVoice and Paraformer models have been tested at this time. Further testing with other model types is encouraged.
(This implementation was developed with the assistance of an AI language model.)

sherpa-onnx offers great performance and is significantly lighter than FunASR. (no torch)

### CODE REVIEW

**Comment by t41372** [comment]
太牛了👍 我来看看
话说能在 `README.md` 里面添加一下文档吗？ (`README.CN.md` 现在跟 `README.md` 没有很同步，我之后会一起更新)

**Comment by t41372** [comment]
我也在看这个 sherpa-onnx，感觉真不错。正想着他的 tts 部分能替代 Melo 你就更新了，太赞了👍。

**Comment by Neil2893** [comment]
已经更新了 README.md，添加了简要的安装使用说明， 虽然觉得sherpa-onnx可以代替 funasr 成为默认的推荐，但还是等等后面大家使用的反馈吧。

**Comment by t41372** [comment]
抱歉，我这几天期末考有点忙来着。我一两天之后再来看...

**Comment by Neil2893** [comment]
> 抱歉，我这几天期末考有点忙来着。我一两天之后再来看...

没关系的，还请先好好准备考试。我能理解是责任感在作怪，但无需感到压力。

File: conf.yaml [hunk 1]
```diff
@@ -143,6 +143,46 @@ FunASR:
...
+  # nemo_ctc: ""        # Path to the NeMo CTC model (e.g., "path/to/model.onnx")
+  # --- For model_type: "wenet_ctc" ---
+  # wenet_ctc: ""       # Path to the WeNet CTC model (e.g., "path/to/model.onnx")
+  # --- For model_type: "tdnn_ctc" ---
+  # tdnn_model: ""      # Path to the TDNN CTC model (e.g., "path/to/model.onnx")
+  # --- For model_type: "whisper" ---
+  # whisper_encoder: "" # Path to the Whisper encoder model (e.g., "path/to/encoder.onnx")
+  # whisper_decoder: "" # Path to the Whisper decoder model (e.g., "path/to/decoder.onnx")
+  # --- For model_type: "sense_voice" ---
+  sense_voice: "/path/to/sherpa-onnx-sense-voice-zh-en-ja-ko-yue-2024-07-17/model.onnx" # Path to the SenseVoice model (e.g., "path/to/model.onnx")
```
Comment (csukuangfj) [inline conf.yaml:168]: I suggest that you use model.int8.onnx, which is way smaller in file size than that of model.onnx.

Comment (Neil2893) [reply conf.yaml:168]: Thanks for the heads up! I appreciate the suggestion to use model.int8.onnx. Do you have any insights into how the int8 model performs compared to the original model.onnx in terms of recognition accuracy? Have you had a chance to evaluate its performance?

Comment (csukuangfj) [reply conf.yaml:168]: I only tested it on fewer than 4 test wave files and the quantized model produces identical results as the not quantized model.

**Pull request merged**